src/components/output/custom-els/MultiPanel/index.ts

```diff
@@ -0,0 +1,208 @@
+  // if children are odd numbers, add a div at the end to prevent potential error.
+  _childrenChange () {
+    if (this.children.length % 2 !== 0) {
+      console.error(`detected odd number of elements inside multi-panel,
+      please make sure you have heading/content pair`);
+    }
+
+    const children : Element[] = Array.from(this.children);
+    let tabindexAlreadySet : boolean = false;
+
+    for (let i = 0; i < children.length; i += 2) {
+      const heading = children[i];
+      const content = children[i + 1];
+      const randomId = Math.random().toString(36).substr(2, 9);
+
+      // if panel-heading/panel-content class is already assigned to elements
+      // (= new child was added earlier), skip the ID/tabindex setting activities
+      // and just flag tabindexAlreadySet
+      if (heading.classList.contains('panel-heading')
+        && content.classList.contains('panel-content')) {
+        tabindexAlreadySet = true;
+        continue;
+      }
+
+      // for A11y add id and tab index
+      heading.classList.add('panel-heading');
+      heading.setAttribute('tabindex', '-1');
+      heading.id = `panel-heading-${randomId}`;
+      heading.setAttribute('aria-controls', `panel-content-${randomId}`);
+
+      content.classList.add('panel-content');
+      content.id = `panel-content-${randomId}`;
+      content.setAttribute('aria-labelledby', `panel-heading-${randomId}`);
+    }
+
+    // Only if tab index was never set (= initial load) make the first heading focusable.
+    // otherwise keep tab index where already is assigned.
+    if (!tabindexAlreadySet) {
+      children[0].setAttribute('tabindex', '0');
+    }
+  }
+
+  _getClosestHeading (el: Element) {
+    const closestEl = el.closest('multi-panel > *');
+    if (closestEl && closestEl.classList.contains('panel-heading')) {
+      return closestEl as HTMLElement;
+    }
+    return undefined;
+  }
+
+  // returns headding that is before currently selected one.
+  _prevHeading () {
+    // activeElement would be the currently selected headding
+    // 2 elemements before that would be the previouse heading unless it is the first element.
+    if (this.firstElementChild === document.activeElement) {
+      return this.firstElementChild as HTMLElement;
+    }
+    // previouse Element of active Element is previouse Content,
+    // previouse Element of previouse Content  is previouseHeadding
+    const previouseContent = document.activeElement.previousElementSibling;
+    if (previouseContent) {
+      return previouseContent.previousElementSibling as HTMLElement;
+    }
+    return;
+  }
+
+  // returns headding that is after currently selected one.
+  _nextHeading () {
+    // activeElement would be the currently selected headding
+    // 2 elemements after that would be the next heading.
+    const nextContent = document.activeElement.nextElementSibling;
+    if (nextContent) {
+      return nextContent.nextElementSibling as HTMLElement;
+    }
+    return;
+  }
```

Comment:
I'd usually ask for this to be a function (as it doesn't use internal state), but it makes sense to leave it so it's symmetrical with `_prevHeading`.

Reply:
oh wait, you answered my question on previous comment ! :)